crates/weaver_live_check/src/sample_span.rs

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Intermediary format for telemetry sample spans
+
+use serde::{Deserialize, Serialize};
+use weaver_semconv::group::SpanKindSpec;
+
+use crate::{
+    live_checker::LiveChecker, sample_attribute::SampleAttribute, LiveCheckResult, LiveCheckRunner,
+    LiveCheckStatistics, SampleRef,
+};
+
+/// Represents a sample telemetry span parsed from any source
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct SampleSpan {
+    /// The name of the span
+    pub name: String,
+    /// The kind of the span
+    pub kind: SpanKindSpec,
+    /// The span's attributes
+    #[serde(default)]
+    pub attributes: Vec<SampleAttribute>,
+    /// SpanEvents
+    #[serde(default)]
+    pub span_events: Vec<SampleSpanEvent>,
+    /// SpanLinks
+    #[serde(default)]
+    pub span_links: Vec<SampleSpanLink>,
+    /// Live check result
+    pub live_check_result: Option<LiveCheckResult>,
+}
+
+impl LiveCheckRunner for SampleSpan {
```

Comment:
Ah, I see, we use this trait to figure out how to nest into possible "sample" things we sent to advice.

I like this.